client/src/modules/stock/entry/entry.js

```diff
@@ -253,7 +261,32 @@ function StockEntryController(Depots, Inventory, Notify,
       user_id     : Session.user.id,
     };
 
-    var lots = vm.Stock.store.data.reduce(function (current, previous) {
+    movement.lots = processLotsFromStore(vm.Stock.store.data, vm.movement.entity.uuid);
+
+    Stock.stocks.create(movement)
+      .then(function (document) {
+        vm.document = document;
+        return Purchase.stockStatus(vm.movement.entity.uuid);
+      })
+      .then(function () {
+        vm.Stock.store.clear();
+        vm.movement = {};
+        ReceiptModal.stockEntryPurchaseReceipt(vm.document.uuid, bhConstants.flux.FROM_PURCHASE);
+      })
+      .catch(Notify.handleError);
+  }
+
+  /**
```

Comment:
Good,  it will be better to precise what are ```data``` and ```uuid``` also in the comments as we have precision about function name, description and return statement